src/mlpack/methods/ann/make_alias.hpp

```diff
@@ -15,44 +15,128 @@
 #define MLPACK_METHODS_ANN_MAKE_ALIAS_HPP
 
 #include <mlpack/prereqs.hpp>
+#include <bandicoot>
 
 namespace mlpack {
 
+template<typename MatType>
+struct IsCootMatType;
+
+template<typename CubeType>
+struct IsCootCubeType;
+
+template<typename MatType>
+struct IsCootMatType
+{
+  constexpr static bool value = false;
+};
+
+template<typename CubeType>
+struct IsCootCubeType
+{
+  constexpr static bool value = false;
+};
+
+#ifdef MLPACK_HAS_COOT
+template<typename MatType>
+struct IsCootMatType
+{
+  constexpr static bool value = true;
+};
+
+template<typename CubeType>
+struct IsCootCubeType
+{
+  constexpr static bool value = true;
+};
```

Comment:
I don't think the strategy here will work correctly.  This returns `true` (and is a duplicate definition) whenever `MLPACK_HAS_COOT` is defined.  I think you need something more like this:

```
template<typename Mattype>
struct IsCootMatType
{
  constexpr static bool value = false;
};

#ifdef MLPACK_HAS_COOT
template<typename eT>
struct IsCootMatType<coot::Mat<eT>>
{
  constexpr static bool value = true;
};
#endif
```

e.g. we should only specialize to `true` specifically for Bandicoot objects.

Reply:
You are definitely right, what you proposed is what was in my brain, I have no idea how did I right it like this :sweat_smile: